app/lib/features/chat/pages/edit_chat_page.dart

```diff
@@ -0,0 +1,218 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/common/widgets/base_body_widget.dart';
+import 'package:acter/common/widgets/input_text_field.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:path/path.dart' as p;
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+import 'package:path_provider/path_provider.dart';
+
+final _log = Logger('a3::chat::edit_profile_page');
+
+class EditChatPage extends ConsumerStatefulWidget {
+  final String roomId;
+
+  const EditChatPage({
+    required this.roomId,
+    super.key,
+  });
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _EditProfilePageState();
+}
+
+class _EditProfilePageState extends ConsumerState<EditChatPage> {
+  final ValueNotifier<File?> selectedUserAvatar = ValueNotifier(null);
+  final chatNameController = TextEditingController();
+  final chatDescriptionController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    _setChatData();
+  }
+
+  Future<void> _setChatData() async {
+    final profileData =
+        await ref.read(chatProfileDataProviderById(widget.roomId).future);
+    final convoData = await ref.read(chatProvider(widget.roomId).future);
+    if (profileData.hasAvatar()) {
+      Directory appDocDir = await getApplicationDocumentsDirectory();
+      final dirPath = p.join(appDocDir.path, 'dir');
+      await Directory(dirPath).create(recursive: true);
+      String filePath = p.join(appDocDir.path, '${widget.roomId}.jpg');
+      final imageFile = File(filePath);
+      imageFile.writeAsBytes(profileData.avatar!.asTypedList());
+      selectedUserAvatar.value = imageFile;
+    }
```

Comment:
why is this not just using the regular `ActerAvatar` facilities?

Reply:
This due to flow. When user select avatar, not uploading it directly and showing updated avatar from local file path and It will be uploaded upon click of Save button.

But let me do needful changes on the flow with inline approach so that this won't be the case.